shell/browser/browser_win.cc

```diff
@@ -584,8 +589,120 @@ v8::Local<v8::Promise> Browser::GetApplicationInfoForProtocol(
   return handle;
 }
 
-bool Browser::SetBadgeCount(int count) {
-  return false;
+int Browser::GetBadgeCount(base::Optional<bool> shouldGetBadge) {
+  if (!shouldGetBadge.has_value() || !shouldGetBadge.value()) {
+    return badge_count_;
+  } else {
+    return 0;
+  }
+}
+
+bool Browser::SetBadgeCount(base::Optional<int> count,
+                            base::Optional<bool> shouldSetBadge) {
+  if (!shouldSetBadge.has_value() || !shouldSetBadge.value()) {
+    // app.setBadgeCount(count) is not support on Windows, but
+    // navigator.setAppBadge is
+    return false;
+  }
+  base::Optional<std::string> badge_content;
+  if (count.has_value() && count.value() == 0) {
+    badge_content = base::nullopt;
+  } else {
+    badge_content = badging::BadgeManager::GetBadgeString(count);
+  }
+
+  // There are 3 different cases when the badge has a value:
+  // 1. |contents| is between 1 and 99 inclusive => Set the accessibility text
+  //    to a pluralized notification count (e.g. 4 Unread Notifications).
+  // 2. |contents| is greater than 99 => Set the accessibility text to
+  //    More than |kMaxBadgeContent| unread notifications, so the
+  //    accessibility text matches what is displayed on the badge (e.g. More
+  //    than 99 notifications).
+  // 3. The badge is set to 'flag' => Set the accessibility text to something
+  //    less specific (e.g. Unread Notifications).
+  std::string badge_alt_string;
+  if (count.has_value()) {
+    badge_count_ = count.value();
+    badge_alt_string = (uint64_t)badge_count_ <= badging::kMaxBadgeContent
+                           // Case 1.
+                           ? l10n_util::GetPluralStringFUTF8(
+                                 IDS_BADGE_UNREAD_NOTIFICATIONS, badge_count_)
+                           // Case 2.
+                           : l10n_util::GetPluralStringFUTF8(
+                                 IDS_BADGE_UNREAD_NOTIFICATIONS_SATURATED,
+                                 badging::kMaxBadgeContent);
+  } else {
+    // Case 3.
+    badge_alt_string =
+        l10n_util::GetStringUTF8(IDS_BADGE_UNREAD_NOTIFICATIONS_UNSPECIFIED);
+    badge_count_ = 0;
+  }
+  for (auto* window : WindowList::GetWindows()) {
+    // Update all the windows
+    HWND hwnd = window->GetAcceleratedWidget();
+    DWORD process_id = 0;
+
+    GetWindowThreadProcessId(hwnd, &process_id);
+    // On Windows set the badge on the first window found for this process.
+    if (process_id == GetCurrentProcessId()) {
```

Comment:
There is no need to get and compare `process_id`, `WindowList` only returns windows for current app.